app/views/teachers/TeacherStudentView.coffee

```diff
@@ -0,0 +1,392 @@
+    @supermodel.trackRequest(@levels.fetchForClassroom(classroomID, {data: {project: 'name,original'}}))
+    #
+    # @user = new User({_id: studentID})
+    # @supermodel.trackRequest(@user.fetch())
+
+    @urls = require('core/urls')
+
+
+    @singleStudentLevelProgressDotTemplate = require 'templates/teachers/hovers/progress-dot-single-student-level'
+    @levelProgressMap = {}
+
+    super(options)
+
+  onLoaded: ->
+    if @students.loaded and not @destroyed
+      @user = _.find(@students.models, (s)=> s.id is @studentID)
+      @updateLastPlayedString()
+      @updateLevelProgressMap()
+      @updateLevelDataMap()
+      @render()
+    super()
+
+  afterRender: ->
+    super(arguments...)
+    $('.progress-dot, .btn-view-project-level').each (i, el) ->
+      dot = $(el)
+      dot.tooltip({
+        html: true
+        container: dot
+      }).delegate '.tooltip', 'mousemove', ->
+        dot.tooltip('hide')
+
+    # @drawLineGraph()
+    @drawBarGraph()
+    @onChangeCourseChart()
+
+
+  onChangeCourseChart: (e)->
+    # show() the one that's selected
+    # console.log ((e.currentTarget).value)
+    if (e)
+      selected = ('#visualisation-'+((e.currentTarget).value))
+      $("[id|='visualisation']").hide()
+      $(selected).show()
+
+
+    # hide() all the others
+
+  drawBarGraph: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @levelData
+
+    WIDTH = 1142
+    HEIGHT = 600
+    MARGINS = {
+      top: 50
+      right: 20
+      bottom: 50
+      left: 70
+    }
+
+    # console.log (@classroom)
+
+    for versionedCourse in @classroom.get('courses') ? []
+      # this does all of the courses, logic for whether student was assigned is in corresponding jade file
+      vis = d3.select('#visualisation-'+versionedCourse._id)
+      courseLevelData = []
+      for level in @levelData when level.courseID is versionedCourse._id
+        courseLevelData.push level
+
+      course = _.find @courses.models, (c) => c.id is versionedCourse._id
+
+      levels = @classroom.getLevels({courseID: course.id}).models
+
+
+      xRange = d3.scale.ordinal().rangeRoundBands([MARGINS.left, WIDTH - MARGINS.right], 0.1).domain(courseLevelData.map( (d) -> d.levelIndex))
+      yRange = d3.scale.linear().range([HEIGHT - (MARGINS.top), MARGINS.bottom]).domain([0, d3.max(courseLevelData, (d) -> if d.class > d.student then d.class else d.student)])
+      xAxis = d3.svg.axis().scale(xRange).tickSize(1).tickSubdivide(true)
+      yAxis = d3.svg.axis().scale(yRange).tickSize(1).orient('left').tickSubdivide(true)
+
+      vis.append('svg:g').attr('class', 'x axis').attr('transform', 'translate(0,' + (HEIGHT - (MARGINS.bottom)) + ')').call xAxis
+      vis.append('svg:g').attr('class', 'y axis').attr('transform', 'translate(' + MARGINS.left + ',0)').call yAxis
+
+      chart = vis.selectAll('rect')
+      .data(courseLevelData)
+      .enter()
+      chart.append('rect')
```

Comment:
I'd add a line comment before each of these `append`s just saying what it's drawing.